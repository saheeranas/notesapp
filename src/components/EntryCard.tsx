import React from 'react';
import {StyleSheet, View, Pressable} from 'react-native';

import {Text} from '@ui-kitten/components';
import dayjs from 'dayjs';

interface EntryCardProps {
  item: any;
  onPress: () => void;
}

const EntryCard: React.FC<EntryCardProps> = ({item: {desc, date}, onPress}) => {
  const dayStr = dayjs(date).format('MMM D h:mm A');
  return (
    <Pressable style={styles.listItem} onPress={onPress}>
      <View style={styles.listItemInner}>
        <Text style={styles.desc}>{desc.substr(0, 50)}</Text>
        <View style={styles.dateWrp}>
          <Text style={styles.dayStr}>{dayStr}</Text>
        </View>
      </View>
    </Pressable>
  );
};

export default EntryCard;

const styles = StyleSheet.create({
  listItem: {
    marginBottom: 10,
    width: '48%',
  },
  listItemInner: {
    paddingHorizontal: 16,
    paddingVertical: 8,
    backgroundColor: '#fff',
    borderRadius: 5,
    alignItems: 'flex-start',
  },
  dateWrp: {
    marginBottom: 3,
    alignItems: 'center',
    borderTopWidth: StyleSheet.hairlineWidth,
    borderTopColor: '#ccc',
    paddingTop: 8,
  },
  dayStr: {
    fontSize: 10,
    color: '#333333',
    textAlign: 'center',
    textTransform: 'uppercase',
  },
  desc: {
    fontSize: 15,
    color: '#333333',
    flex: 1,
    maxHeight: 40,
    paddingBottom: 8,
  },
});
