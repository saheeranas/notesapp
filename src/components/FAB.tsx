import React from 'react';
import {StyleSheet} from 'react-native';
import {Button, Icon, ButtonProps} from '@ui-kitten/components';

interface FABProps extends ButtonProps {
  onPress: () => void;
}

const AddIcon = () => (
  <Icon name="plus-outline" style={styles.icon} fill="#fff" />
);

const FAB = (props: FABProps) => {
  return (
    <Button
      style={styles.fab}
      {...props}
      onPress={() => props.onPress()}
      accessoryLeft={AddIcon}
      status="warning"
      testID="fab_button"
    />
  );
};

const styles = StyleSheet.create({
  fab: {
    width: 46,
    height: 46,
    borderRadius: 46 / 2,
    borderWidth: 0,
    position: 'absolute',
    elevation: 1,
    zIndex: 1,
    bottom: 16,
    right: 16,
    padding: 0,
  },
  icon: {width: 24, height: 24},
});

export default FAB;
