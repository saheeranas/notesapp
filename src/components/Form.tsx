import React, {forwardRef} from 'react';
import {TextInput, StyleSheet, View} from 'react-native';
import {Input, InputProps} from '@ui-kitten/components';

interface SearchProps extends InputProps {}

export const TextArea = forwardRef((props: any, ref) => {
  return <TextInput ref={ref} style={styles.textarea} {...props} />;
});

export const Search = (props: SearchProps) => {
  return (
    <View style={styles.searchCard}>
      <Input style={styles.searchInput} {...props} />
    </View>
  );
};

const styles = StyleSheet.create({
  textarea: {
    borderWidth: 0,
  },
  // Search
  searchCard: {
    backgroundColor: 'transparent',
    paddingHorizontal: 16,
    paddingTop: 0,
    paddingBottom: 3,
  },
  searchInput: {
    backgroundColor: '#ececec',
  },
});
