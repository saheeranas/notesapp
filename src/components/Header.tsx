import React from 'react';
import {StyleSheet} from 'react-native';
import {
  Icon,
  TopNavigation,
  TopNavigationAction,
  Text,
} from '@ui-kitten/components';

import {HeaderType} from '../types/types';

const BackIcon = () => (
  <Icon style={{width: 20, height: 20}} name="arrow-back" fill="#000" />
);

const KebabIcon = () => (
  <Icon
    style={{width: 20, height: 20}}
    name="more-vertical-outline"
    fill="#000"
  />
);

const Header: React.FC<HeaderType> = ({
  title = 'Diary',
  navigation,
  hideBack = false,
  style,
  onPressMenu,
}) => {
  const BackAction = () => (
    <TopNavigationAction icon={BackIcon} onPress={() => navigation.goBack()} />
  );

  const MenuButton = () => (
    <TopNavigationAction
      icon={KebabIcon}
      onPress={() => {
        onPressMenu && onPressMenu();
      }}
    />
  );

  const Title = () => <Text category="s1">{title}</Text>;

  return (
    <TopNavigation
      accessoryLeft={hideBack ? undefined : BackAction}
      accessoryRight={onPressMenu ? MenuButton : undefined}
      title={Title}
      alignment="center"
      style={[styles.header, style]}
    />
  );
};

const styles = StyleSheet.create({
  header: {
    backgroundColor: 'transparent',
  },
});

export default Header;
