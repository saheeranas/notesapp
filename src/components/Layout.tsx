import React, {ReactNode} from 'react';
import {StyleSheet} from 'react-native';
import {Layout as UILayout} from '@ui-kitten/components';

interface LayoutProps {
  children?: ReactNode;
}

export const Layout = ({children}: LayoutProps) => {
  return <UILayout style={styles.layout}>{children}</UILayout>;
};

const styles = StyleSheet.create({
  safearea: {
    flex: 1,
  },
  layout: {
    position: 'relative',
    flex: 1,
    backgroundColor: '#F4F4F4',
  },
});
