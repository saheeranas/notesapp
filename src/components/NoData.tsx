import React from 'react';
import {StyleSheet, View} from 'react-native';
import {Text} from '@ui-kitten/components';

interface NoDataProps {
  title?: string;
}

const NoData = ({title = 'No Data'}: NoDataProps) => {
  return (
    <View style={styles.wrapper}>
      <Text status="basic" category="p2">
        {title}
      </Text>
    </View>
  );
};

export default NoData;

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
