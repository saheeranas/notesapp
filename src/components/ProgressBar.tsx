import React from 'react';
import {StyleSheet, View} from 'react-native';

interface ProgressBarProps {
  color: string;
  progress: number;
}

const ProgressBar = ({color = '#34c759', progress = 0}) => {
  return (
    <View style={styles.wrapper}>
      <View
        style={[
          styles.bar,
          {width: `${progress * 100}%`, backgroundColor: color},
        ]}
      />
    </View>
  );
};

export default ProgressBar;

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: '#ccc',
    borderRadius: 15,
  },
  bar: {
    backgroundColor: '#34c759', //green
    height: 8,
    borderRadius: 15,
  },
});
