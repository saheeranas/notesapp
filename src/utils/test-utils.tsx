import * as React from 'react';
import {render} from '@testing-library/react-native';
import * as eva from '@eva-design/eva';
import {ApplicationProvider, IconRegistry} from '@ui-kitten/components';
import {EvaIconsPack} from '@ui-kitten/eva-icons';

// Navigation
// import RootNavigation from '../routes/RootNavigation';

const AllTheProviders = ({children}) => {
  return (
    <>
      <IconRegistry icons={EvaIconsPack} />
      <ApplicationProvider {...eva} theme={eva.light}>
        {children}
      </ApplicationProvider>
    </>
  );
};

const customRender = (ui, options) =>
  render(ui, {wrapper: AllTheProviders, ...options});

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
