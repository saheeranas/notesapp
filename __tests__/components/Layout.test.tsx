import 'react-native';
import {View} from 'react-native';
import React from 'react';
import {render} from '../../src/utils/test-utils';

import {Layout} from '../../src/components/Layout';

it('renders Layout', () => {
  const tree = render(<Layout />, {}).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Layout with children', () => {
  render(
    <Layout>
      <View>
        <View />
      </View>
    </Layout>,
    {},
  );
});
