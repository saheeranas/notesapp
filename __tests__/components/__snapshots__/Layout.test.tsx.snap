// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Layout 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    appearance="default"
    level="1"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
        },
        Object {
          "backgroundColor": "#F4F4F4",
          "flex": 1,
          "position": "relative",
        },
      ]
    }
  />
</View>
`;
